AmplifyPlugins/Core/AWSPluginsCore/Model/Support/ModelSchema+GraphQL.swift

```diff
@@ -11,9 +11,26 @@ import Foundation
 /// Extension that adds GraphQL specific utilities to `ModelSchema`.
 extension ModelSchema {
 
-    /// The GraphQL name of the schema.
-    var graphQLName: String {
-        name
+    /// The GraphQL directive name translated from the GraphQL operation and model schema data
+    func graphQLName(operationSubType: GraphQLOperationSubType) -> String {
```

Comment:
This feels like an appropriate use of an extension: taking a custom type and extending it with more custom behavior that doesn't necessarily represent the core functionality of the type.

Reply:
ended up removing `GraphQLOperationSubType` and creating three extensions with parameters taking in `GraphQLQueryType' `GraphQLMutationType, etc